api/app/Models/PoolCandidate.php

```diff
@@ -411,4 +411,29 @@ public function isDraft()
     {
         return is_null($this->submitted_at) || $this->submitted_at->isFuture();
     }
+
+    /**
+     * Scope the query to PoolCandidate's the current user can view
+     */
+    public function scopeAvailableToQuery(Builder $query)
```

Comment:
Does this cover the case where a user is trying to view their own application?

Reply:
I had that and removed it for some reason 🤔 Added back in [update scope to filter draft, restrict viewing own](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6064/commits/4b95d1f03c3c21c8d35155b715741e7fcb796fbd)